plugins/modules/cloud/misc/proxmox_kvm.py

```diff
@@ -960,7 +960,13 @@ def create_vm(self, vmid, newid, node, name, memory, cpu, cores, sockets, update
                 kwargs.update(kwargs[k])
                 del kwargs[k]
 
-        # Rename numa_enabled to numa. According the API documentation
+        # Map bool to string, according to API documentation.
+        if kwargs['agent'] is True:
```

Comment:
`agent` is a string argument, so anything passed in will be converted to a string. So now you have to check for *anything* that Ansible accepts as a truish boolean.

You can use something like
```
from ansible.module_utils.parsing.convert_bool import boolean

...

try:
    if boolean(kwargs['agent'], strict=True):
        kwargs['agent'] = 'enabled=1'
    else:
        kwargs['agent'] = 'enabled=0'
except TypeError:
    # Not something that Ansible would parse as a boolean
    pass
```
to handle this correctly.

Reply:
Thanks, I added this now.